src/ceph.in

```diff
@@ -899,7 +899,10 @@ def main():
     try:
         if childargs and childargs[0] == 'ping':
             return ping_monitor(cluster_handle, childargs[1], timeout)
-        run_in_thread(cluster_handle.connect, timeout=timeout)
+        result = run_in_thread(cluster_handle.connect, timeout=timeout)
+        if type(result) is tuple and result[0] == -errno.EINTR:
```

Comment:
I'd just as soon regularize the return from run_in_thread to a tuple, then, to reduce confusion and verbosity of test.

a long time ago my intent was to propagate the KeyboardInterrupt as the "the thread was interrupted either by human or timeout".  I don't know when that changed (could have been before I pushed originally) but it still seems like a plausible approach.